shub/utils.py

```diff
@@ -476,6 +472,16 @@ def latest_github_release(force_update=False, timeout=1., cache=None):
     return release_data
 
 
+def read_release_data(cache):
+    """Read release data from file. Returns empty dict if data is missing."""
+    with open(cache, 'r') as f:
+        try:
+            release_data = json.load(f)
+        except ValueError:
+            release_data = {}
+    return release_data
+
```

Comment:
I see that I screwed up the error handling in the first place. What I meant to do (and why I used `catch Exception`) is "don't fail when the file can for some reason not be read", i.e. "never fail":
```python
    try:
        with open(cache, 'r') as f:
            release_data = json.load(f)
    except Exception:
        release_data = {}
    return release_data
```